### PR TITLE
Fix deprecation for java.lang.Deprecated

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

File: scala3doc/src/dotty/dokka/tasty/BasicSupport.scala [hunk 1]
```diff
@@ -44,6 +44,12 @@ trait BasicSupport:
     def getAnnotations(): List[Annotation] =
       sym.annotations.filterNot(_.symbol.packageName.startsWith("scala.annotation.internal")).map(parseAnnotation).reverse
 
+    def isDeprecated(): Option[Annotation] =
+      sym.annotations.find { a =>
+        a.symbol.packageName == "scala" && a.symbol.className.fold(false)(_ == "deprecated") ||
```
Comment (romanowski) [inline scala3doc/src/dotty/dokka/tasty/BasicSupport.scala:49]: `fold(false)(_ == "deprecated")` can be replaced with '`.contains("deprecated")"`

**Review by romanowski — APPROVED** [review]

File: scala3doc/src/dotty/dokka/tasty/BasicSupport.scala (hunk 1 above)
Comment (BarkingBad) [reply scala3doc/src/dotty/dokka/tasty/BasicSupport.scala]: Changes

**Comment by griggt** [comment]
This has caused CI breakage on JDK 8

https://github.com/lampepfl/dotty/runs/1811701716?check_suite_focus=true#step:8:142

```
Error:  -- Error: /__w/dotty/dotty/scala3doc-testcases/src/tests/deprecated.scala:22:28 
Error:  22 |@java.lang.Deprecated(since = "1.1")
Error:     |                      ^^^^^^^^^^^^^
Error:     |constructor Deprecated in class Deprecated: (): Deprecated does not have a parameter since
Error:  one error found
```

**Comment by BarkingBad** [comment]
Sorry for the trouble, didn't see it coming as PR had passed all smaller checks

**Comment by griggt** [comment]
#11280

**Comment by nicolasstucki** [comment]
Tho avoid these kinds of failures it is good to use Java 8 on the Dotty project. If it works with 8 it most likely works with later versions we support.

**Pull request merged**